drivers/soundwire/master.c

```diff
@@ -142,10 +142,11 @@ int sdw_master_device_add(struct sdw_bus *bus, struct device *parent,
 	md->dev.parent = parent;
 	md->dev.groups = master_node_groups;
 	md->dev.of_node = parent->of_node;
-	md->dev.fwnode = fwnode;
 	md->dev.dma_mask = parent->dma_mask;
 
 	dev_set_name(&md->dev, "sdw-master-%d", bus->id);
+	/* make sure ACPI companion is set based on parent */
+	ACPI_COMPANION_SET(&md->dev, ACPI_COMPANION(parent));
```

Comment:
I don't know if this is really necessary?
Do we hit any issues without this?

Also this wouldn't work for Qualcomm (not using ACPI), I think we have to use set_primary_fwnode() directly.